### ISSUE TITLE
[Bug] Tales of Symphonia - Black Screen on Fast Travel (Palmacosta to Hakonesia Peak)

### ISSUE THREAD

**Issue by cpawliuk** [FIX94/Nintendont#1191]
If you're in Palmacosta and talk to the person who wants you to guide them to Hakonesia Peak, but say no, and then talk again but say yes you'll get to that location with a black screen and music playing but controls locked. I've seen it posted a few times online about the black screen with the escorting part but just re-produced it to happen every time this way. As long as you always say yes it never triggers the black screen in my tests thus far.

I tested this in Dolphin just to confirm it wasn't a game bug and it doesn't happen, only with Nintendont (using the latest build).

Dump has been verified 100% as well.

![image](https://github.com/FIX94/Nintendont/assets/61577065/13bceb58-a1ad-469f-bfab-66260fd33f38)

Here is a copy of the save file (use Slot A and the first save). Exit the Inn and go to the left and talk to the person near the exit to the city.

[GQSE.zip](https://github.com/FIX94/Nintendont/files/14879923/GQSE.zip)

**Comment by SuperrSonic** [comment]
It's the ISO cache.

**Comment by cpawliuk** [comment]
> It's the ISO cache.

I'm not aware of any cache setting in Nintendont, at least in the latest release if that is what you're asking.

With Dolphin I just kept everything on default when testing.

Did the issue also happen for you with the test save?

**Comment by SuperrSonic** [comment]
It's not a setting, it's a core "feature" in nintendont. Whatever MB are left of MEM2 is used by nintendont to cache game files, if the game requests the file again and it's in the cache you get instant loading (and you also get a ton of problems!)

This is what I tested:
- I loaded ToS in my personal pre-bba fork and couldn't get it to hang at all.
- I loaded ToS in the latest build of nintendont and the game hanged exactly as you described.
- I built my fork with the ISO cache enabled and it hanged exactly as you described.

My conclusion is it's 100% caused by the ISO cache.

Here's the amazing backstory:
I don't use the ISO cache because it randomly breaks music in Melee and every game that uses .hps audio files.

I tried presenting this idea to the devs a long time ago but if you enabled disc read speed emulation the audio glitches would not trigger. So it was ignored, that wasn't good enough for me because nintendont is considerably slower than Devolution and DM/DML, if you enable disc read speed emulation _you can get slower results than a real disc_.

Fast-forward some years, Fire Emblem was found to be unstable when loading chapters.

I try to push again to remove the ISO cache, [my changes finally get through](https://github.com/FIX94/Nintendont/commit/eae81294b4f701612f2f467676310c54c614e5c2)... until [3 years ago](https://github.com/FIX94/Nintendont/commit/6f8bf7348eeafa96a91111478b5bad52f82ae4c8) where it got deliberately enabled again, _except_ for Fire Emblem. His only comment being "Lots of games had random freezes without iso cache."

Unfortunately, I don't seem to possess any of the "lots of games" he tested because I could never get anything to freeze.

**Comment by cpawliuk** [comment]
@SuperrSonic Thank you. I took a look and I see what was adjusted throughout the revisions so I made my own edit.

I attempted to build Nintendont and sadly I'm stuck here, any ideas?

```
Building Nintendont PADReadGC

C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe -C loader/source/p
pc/PADReadGC
make[1]: Entering directory 'C:/Nintendont-master/loader/source/ppc/PADReadGC'
PADReadGC.c
_crt0.S
linking ... PADReadGC.elf
C:/Nintendont-master/loader/source/ppc/PADReadGC/link.ld:18 cannot move location
 counter backwards (from 93003284 to 93003000)
collect2: ld returned 1 exit status
make[2]: *** [C:\devkitpro\devkitPPC/wii_rules:25: C:/Nintendont-master/loader/s
ource/ppc/PADReadGC/PADReadGC.elf] Error 1
make[1]: *** [Makefile:67: build] Error 2
make[1]: Leaving directory 'C:/Nintendont-master/loader/source/ppc/PADReadGC'
make: *** [Makefile:69: loader/source/ppc/PADReadGC] Error 2
```

Not sure how to get passed this: **link.ld:18 cannot move location
 counter backwards (from 93003284 to 93003000)**
 
 Here is link.Id (The 18th line when you download the source is just a line break at the bottom):
 https://github.com/FIX94/Nintendont/blob/master/loader/source/ppc/PADReadGC/link.ld
 
 I'm using Cygdrive.
 
 If I try to run 'make forced' through Powershell or CMD it throws:
 
 ```
" "
"Cleaning all subprojects..."
" "
C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe -C multidol clean
make[1]: Entering directory 'C:/Nintendont-master/multidol'
rm -f -rf multidol_ldr.elf ../loader/data/multidol_ldr.bin crt0.o cache.o main.o global.o apploader.o dip.o utils.o .deps
process_begin: CreateProcess(NULL, rm -f -rf multidol_ldr.elf ../loader/data/multidol_ldr.bin crt0.o cache.o main.o global.o apploader.o dip.o utils.o .deps, ...) failed.
make (e=2): The system cannot find the file specified.
make[1]: [Makefile:51: clean] Error 2 (ignored)
make[1]: Leaving directory 'C:/Nintendont-master/multidol'
C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe -C kernel/asm clean
make[1]: Entering directory 'C:/Nintendont-master/kernel/asm'
clean ...
process_begin: CreateProcess(NULL, rm -fr *.d *.h *.bin *.elf, ...) failed.
make (e=2): The system cannot find the file specified.
make[1]: *** [Makefile:45: clean] Error 2
make[1]: Leaving directory 'C:/Nintendont-master/kernel/asm'
make: *** [Makefile:94: clean] Error 2
```

**Comment by cpawliuk** [comment]
@SuperrSonic Thanks! I'll just add the changes as I already when through the files to see. I'm surprised there isn't a toggle option for this though to at least let the user decide.

Sadly I cannot complete the build... I'm using the tools as per the READ ME. I even had to use Cygdrive on Windows due to commands like uname. Apart from that I couldn't even get half way through using either Powershell or CMD due to several failed processes or something not being found. Almost made it to the last part but I'm stuck here.

```
Building Nintendont loader

C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe -C loader
make[1]: Entering directory 'C:/Nintendont-master/loader'
IOSInterface.bin
/usr/bin/sh: line 1: bin2s: command not found
PADReadGC.bin
/usr/bin/sh: line 1: bin2s: command not found
background.png
/usr/bin/sh: line 1: bin2s: command not found
font.zip
/usr/bin/sh: line 1: bin2s: command not found
kernel.zip
/usr/bin/sh: line 1: bin2s: command not found
kernelboot.bin
/usr/bin/sh: line 1: bin2s: command not found
multidol_ldr.bin
/usr/bin/sh: line 1: bin2s: command not found
stub.bin
/usr/bin/sh: line 1: bin2s: command not found
FPad.c
In file included from C:/Nintendont-master/loader/include/FPad.h:24,
                 from C:/Nintendont-master/loader/source/FPad.c:22:
C:/Nintendont-master/loader/include/global.h:24:10: fatal error: gctypes.h: No such file or directory
 #include <gctypes.h>
          ^~~~~~~~~~~
compilation terminated.
make[2]: *** [C:\devkitpro\devkitPPC/base_rules:52: FPad.o] Error 1
make[1]: *** [Makefile:108: build] Error 2
make[1]: Leaving directory 'C:/Nintendont-master/loader'
make: *** [Makefile:87: loader] Error 2
```

So I went into the libogc folder and added the includes as needed. Not sure where I'm supposed to put them otherwise, and the missing libraries into loader\extlibs\lib as per the Makefile under loader as it shows:

```
#---------------------------------------------------------------------------------
# any extra libraries we wish to link with the project
#---------------------------------------------------------------------------------
LIBS	:= -lfreetype -lpngu -lpng -lz \
	   -lwiiuse -lwiidrc -lwupc -lbte -lfatfs-ppc -logc -ldi -lm
```

After fixing that I'm loaded up with more...

```
Building Nintendont loader

C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe -C loader
make[1]: Entering directory 'C:/Nintendont-master/loader'
linking ... loader.elf
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: FPad.o: in function `FPAD_Init':
C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: global.o: in function `Initialise':
C:/Nintendont-master/loader/source/global.c:187: undefined reference to `font_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: global.o: in function `unzip_data':
C:/Nintendont-master/loader/source/global.c:158: undefined reference to `font_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/global.c:158: undefined reference to `font_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/global.c:158: undefined reference to `font_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: global.o: in function `Initialise':
C:/Nintendont-master/loader/source/global.c:190: undefined reference to `background_png'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/global.c:190: undefined reference to `background_png'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: main.o: in function `main':
C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [C:\devkitpro\devkitPPC/wii_rules:25: C:/Nintendont-master/loader/loader.elf] Error 1
make[1]: *** [Makefile:108: build] Error 2
make[1]: Leaving directory 'C:/Nintendont-master/loader'
make: *** [Makefile:87: loader] Error 2

```

You would think from the main page you just download the source, get the files needed for devkitpro, setup your environment variables, and as long as you have make and everything else working just run **make forced** and you're good. But it seems like a big cluster of chaos. I checked those files, they have the headers, but not sure why I cannot resolve this.

**Comment by carnage702** [comment]
> It's not a setting, it's a core "feature" in nintendont. Whatever MB are left of MEM2 is used by nintendont to cache game files, if the game requests the file again and it's in the cache you get instant loading (and you also get a ton of problems!)
> 
> This is what I tested:
> 
>     * I loaded ToS in my personal pre-bba fork and couldn't get it to hang at all.
> 
>     * I loaded ToS in the latest build of nintendont and the game hanged exactly as you described.
> 
>     * I built my fork with the ISO cache enabled and it hanged exactly as you described.
> 
> 
> My conclusion is it's 100% caused by the ISO cache.
> 
> Here's the amazing backstory: I don't use the ISO cache because it randomly breaks music in Melee and every game that uses .hps audio files.
> 
> I tried presenting this idea to the devs a long time ago but if you enabled disc read speed emulation the audio glitches would not trigger. So it was ignored, that wasn't good enough for me because nintendont is considerably slower than Devolution and DM/DML, if you enable disc read speed emulation _you can get slower results than a real disc_.
> 
> Fast-forward some years, Fire Emblem was found to be unstable when loading chapters.
> 
> I try to push again to remove the ISO cache, [my changes finally get through](https://github.com/FIX94/Nintendont/commit/eae81294b4f701612f2f467676310c54c614e5c2)... until [3 years ago](https://github.com/FIX94/Nintendont/commit/6f8bf7348eeafa96a91111478b5bad52f82ae4c8) where it got deliberately enabled again, _except_ for Fire Emblem. His only comment being "Lots of games had random freezes without iso cache."
> 
> Unfortunately, I don't seem to possess any of the "lots of games" he tested because I could never get anything to freeze.

I reverted that change because sonic Ridders extra mode or whatever mode it was would freeze without iso cache, terminator 3 would crash every couple minutes too and even re4 would crash much more often.

**Comment by cpawliuk** [comment]
> > It's not a setting, it's a core "feature" in nintendont. Whatever MB are left of MEM2 is used by nintendont to cache game files, if the game requests the file again and it's in the cache you get instant loading (and you also get a ton of problems!)
> > This is what I tested:
> > ```
> > * I loaded ToS in my personal pre-bba fork and couldn't get it to hang at all.
> > 
> > * I loaded ToS in the latest build of nintendont and the game hanged exactly as you described.
> > 
> > * I built my fork with the ISO cache enabled and it hanged exactly as you described.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > My conclusion is it's 100% caused by the ISO cache.
> > Here's the amazing backstory: I don't use the ISO cache because it randomly breaks music in Melee and every game that uses .hps audio files.
> > I tried presenting this idea to the devs a long time ago but if you enabled disc read speed emulation the audio glitches would not trigger. So it was ignored, that wasn't good enough for me because nintendont is considerably slower than Devolution and DM/DML, if you enable disc read speed emulation _you can get slower results than a real disc_.
> > Fast-forward some years, Fire Emblem was found to be unstable when loading chapters.
> > I try to push again to remove the ISO cache, [my changes finally get through](https://github.com/FIX94/Nintendont/commit/eae81294b4f701612f2f467676310c54c614e5c2)... until [3 years ago](https://github.com/FIX94/Nintendont/commit/6f8bf7348eeafa96a91111478b5bad52f82ae4c8) where it got deliberately enabled again, _except_ for Fire Emblem. His only comment being "Lots of games had random freezes without iso cache."
> > Unfortunately, I don't seem to possess any of the "lots of games" he tested because I could never get anything to freeze.
> 
> I reverted that change because sonic Ridders extra mode or whatever mode it was would freeze without iso cache, terminator 3 would crash every couple minutes too and even re4 would crash much more often.

Thank you for your reply. I'll just adjust if needed when I find a game that has a problem since I cannot toggle it otherwise. The issue is I cannot get the build to work as per my last post. Would you be able to provide any insight? I still cannot get passed this part:

I'm using devkitARM r53-1, devkitPPC r35-2, and libOGC 1.8.23-1

```
Building Nintendont loader

C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe -C loader
make[1]: Entering directory 'C:/Nintendont-master/loader'
linking ... loader.elf
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: FPad.o: in function `FPAD_Init':
C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/FPad.c:58: undefined reference to `PADReadGC_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: global.o: in function `Initialise':
C:/Nintendont-master/loader/source/global.c:187: undefined reference to `font_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: global.o: in function `unzip_data':
C:/Nintendont-master/loader/source/global.c:158: undefined reference to `font_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/global.c:158: undefined reference to `font_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/global.c:158: undefined reference to `font_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: global.o: in function `Initialise':
C:/Nintendont-master/loader/source/global.c:190: undefined reference to `background_png'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/global.c:190: undefined reference to `background_png'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: main.o: in function `main':
C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:686: undefined reference to `kernel_zip'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:692: undefined reference to `kernelboot_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1545: undefined reference to `IOSInterface_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1550: undefined reference to `stub_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin_size'
c:/devkitpro/devkitppc/bin/../lib/gcc/powerpc-eabi/8.3.0/../../../../powerpc-eabi/bin/ld.exe: C:/Nintendont-master/loader/source/main.c:1674: undefined reference to `multidol_ldr_bin'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [C:\devkitpro\devkitPPC/wii_rules:25: C:/Nintendont-master/loader/loader.elf] Error 1
make[1]: *** [Makefile:108: build] Error 2
make[1]: Leaving directory 'C:/Nintendont-master/loader'
make: *** [Makefile:87: loader] Error 2
```

I checked those .c files and the headers are there, and within them there are indeed references like FPad.c has the include for PADReadGC_bin.h and that header has PADReadGC_bin_size. The same when going down the list for the other problems.

Thank you.

**Comment by cpawliuk** [comment]
Well in the mean time I just decide to edit the Binary directly and swap in Tales of Symphonia. So it works now. I would still prefer to actually build this properly so I don't need to do this in future.

[boot.zip](https://github.com/FIX94/Nintendont/files/14904818/boot.zip)

Using this with that save I provided in my first post works now and zero crash.

Any help to get this going would be greatly appreciated.

**Comment by cpawliuk** [comment]
Just a final update on this as I was caught up with some other projects... I was able to build with the current DevKitPro Tools, and added in a toggle option for ISO Caching to the menu. Tested it with the game and when ON it crashes as per the latest build, when OFF it runs as per my prior Binary edit. There are some minor issues like not being able to use the (R + Z + B + D-Pad->DOWN) to return, but I haven't looked at it further. As is this works perfect for what I need and I can just toggle it ON or OFF.

![image](https://github.com/FIX94/Nintendont/assets/61577065/b08b3653-164d-4e9c-873b-93f37569a13e)
